src/util/state-management.spec.js

```diff
@@ -0,0 +1,549 @@
+import assert from 'assert';
+import _ from 'lodash';
+import sinon from 'sinon';
+import React from 'react';
+import { mount } from 'enzyme';
+
+import {
+	getDeepPaths,
+	omitFunctionPropsDeep,
+	bindReducerToState,
+	bindReducersToState,
+	getStatefulPropsContext,
+	buildStatefulComponent
+} from './state-management';
+
+describe('#getDeepPaths', () => {
+	it('should return an empty array when arg is empty object, null, or undefined', () => {
+		assert(_.isEqual([], getDeepPaths({})));
+		assert(_.isEqual([], getDeepPaths()));
+		assert(_.isEqual([], getDeepPaths(null)));
+	});
+
+	it('should return an array of paths for each node with non-plain object value if arg is object', () => {
+		const pagedTableObj = {
+			rows: ['data0', 'data1'],
+			paginator: {
+				selectedPageIndex: 0,
+				selectedPageSize: 10,
+				dropselector: {
+					selectedIndex: 1,
+					options: [5, 10, 20]
+				}
+			}
+		};
+
+		const deepPaths = getDeepPaths(pagedTableObj);
+		const xorPaths = _.xorWith(deepPaths, [
+			['rows'],
+			['paginator', 'selectedPageIndex'],
+			['paginator', 'selectedPageSize'],
+			['paginator', 'dropselector', 'selectedIndex'],
+			['paginator', 'dropselector', 'options']
+		], _.isEqual)
+		assert(_.isEqual([], xorPaths))
+	});
+
+	it('should return an array of paths for each node with non-plain object value if arg is array', () => {
+		const deepPaths = getDeepPaths(['zero', { one: 1 }, 2]);
+		const xorPaths = _.xorWith(deepPaths, [
+			[0],
+			[1, 'one'],
+			[2]
+		], _.isEqual)
+		assert(_.isEqual([], xorPaths))
+	});
+});
+
+describe('#omitFunctionPropsDeep', () => {
+	it('should return an empty object when arg is empty object, null, or undefined', () => {
+		assert(_.isEqual({}, omitFunctionPropsDeep({})));
+		assert(_.isEqual({}, omitFunctionPropsDeep(null)));
+		assert(_.isEqual({}, omitFunctionPropsDeep()));
+	});
+
+	it('should transform to object without function properties', () => {
+		const pagedTableObj = {
+			rows: ['data0', 'data1'],
+			onRowSelect: _.noop,
+			paginator: {
+				selectedPageIndex: 0,
+				selectedPageSize: 10,
+				onPageSizeSelect: _.noop,
+				onPageSelect: _.noop,
+				dropselector: {
+					selectedIndex: 1,
+					options: [5, 10, 20],
+					onSelect: _.noop
+				}
+			}
+		};
+
+		const result = omitFunctionPropsDeep(pagedTableObj);
+
+		assert(_.isEqual(result, {
+			rows: ['data0', 'data1'],
+			paginator: {
+				selectedPageIndex: 0,
+				selectedPageSize: 10,
+				dropselector: {
+					selectedIndex: 1,
+					options: [5, 10, 20]
+				}
+			}
+		}));
+	});
+});
+
+describe('#bindReducerToState', () => {
+	it('should bind a single reducer function to a state management interface', () => {
+		let state = {
+			value: null
+		};
+
+		const stateManager = {
```

Comment:
I _could_ DRY them up a bit by adding `let state;` up to the shared scope and then have stateManager live with it, while each test assigns `state` a specific value individually. But I feel it takes away from isolation and readability of each test by segregating interdependent data. Generally, I feel tests can be too DRY for that reason.